auth/rest_response_parser.rl

```diff
@@ -0,0 +1,148 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright (C) 2015 Cloudius Systems, Ltd.
+ */
+
+// Added to manage status and enforce all header fields to be in lowercase
+#include <seastar/core/ragel.hh>
+#include <memory>
+#include <unordered_map>
+
+namespace auth {
+
+struct http_response {
+    sstring _version;
+    int _status;
+    std::unordered_map<sstring, sstring> _headers;
+};
+
+%% machine http_response;
+
+%%{
+
+access _fsm_;
+
+action mark {
+    g.mark_start(p);
+}
+
+action store_version {
+    _rsp->_version = str();
+}
+
+action store_status {
+    _rsp->_status = std::stoi(str());
+}
+
+action store_field_name {
+    _field_name = str();
```

Comment:
For later, we could store field_name in lower case to avoid case insensitive lookup later

Reply:
Already done see previous comment